torchutils/models.py

```diff
@@ -18,10 +22,28 @@ def get_model_param_count(model):
     """
 
     _validate_param(model, 'model', 'model')
+    _validate_param(param_type, 'param_type', 'str')
+
     param_count = 0
     for p in model.parameters():
-        val = p.size(0)
-        if len(p.size()) > 1:
-            val *= p.size(1)
-        param_count += val
-    return param_count
+        if param_type == 'total':
```

Comment:
Seems like too much code duplication. You can just have single calculation and then return the value based on `param_type`.

```
total_params = 0
trainable_params = 0
for p in model.parameters():
    val = p.size(0)
    if len(p.size()) > 1:
        val *= p.size(1)
    if p.requires_grad:
        trainable_params += val
    total_param += val
````